### ISSUE TITLE
Deep Learning of Binary Hash Codes for Fast Image Retrieval

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Reading-Paper#78]
https://www.iis.sinica.edu.tw/~kevinlin311.tw/cvprw15.pdf

**Comment by chullhwan-song** [comment]
## Abstract
* 간단 리뷰
* large-scale image retrieval
* How 
    * to generate binary hash codes for fast image retrieval. 
    * hamming distance 같은 써서..그렇기 위해서는, CNN FC feature를 hash화 한다. 
    * deep hashing methods        
        * 이를 위해 가장 last layer에서 latent layer(for hash) 둠.
## Method        
![image](https://user-images.githubusercontent.com/40360823/52405894-c15b4a00-2b0f-11e9-8883-4bd7dfbd5512.png)
* 이젠 고전적이긴 하지만, 이게 2015년 논문이니.. FC feature를 이용하여 Similar search를 이미 꽤 좋은 성능이 보인다는 것은 이제 많이 알려져 있다.
* 위의 그림에서, 
    * 먼저, imagenet으로 학습 모델을 만들고.
    * transfer learning을한다. 보통 target domain에 대하여, fine-tuned..라고 많이 부르는..
    * 근데 여기서 Fig.1의 b에서 보면 FC layer다음에 하나더 latent layer라는것이 하나 있다. 
    * sigmoid activation function을 가져 0, 1로 만들어 hash화 한다... (참고 - tanh도 많이 사용하는것으로도 보임)
     * 중요한것은 이는 learning을 통해 한다는것.
     * 여기서는 softmax loss기반이다.(위의 그림을 보면 ...금방알수 있다)
     * 이후,  laten layer에서 만든 hash를 이용하여 hamming distance를 통해 빠르게 retrieval할 수 있다는의미.
* 사실 너무 간단한데..그당시에는..모든게 ..
## 실험
![image](https://user-images.githubusercontent.com/40360823/52406335-d08ec780-2b10-11e9-9cb9-e4381667a7ff.png)
![image](https://user-images.githubusercontent.com/40360823/52406552-56127780-2b11-11e9-9947-b3c8660bdfde.png)
![image](https://user-images.githubusercontent.com/40360823/52406363-df757a00-2b10-11e9-8645-ee48d4037bff.png)
![image](https://user-images.githubusercontent.com/40360823/52406401-f6b46780-2b10-11e9-8288-fb7dd35a30f5.png)
![image](https://user-images.githubusercontent.com/40360823/52406513-45fa9800-2b11-11e9-8fa7-002db1a85851.png)
![image](https://user-images.githubusercontent.com/40360823/52406422-03d15680-2b11-11e9-847a-18a1518ac0a2.png)
![image](https://user-images.githubusercontent.com/40360823/52406446-0f248200-2b11-11e9-855d-84166d366007.png)
